File: src/main/java/com/example/cqrs/command/entity/event/metadata/EventSchemaVersion.java
package com.example.cqrs.command.entity.event.metadata;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

// 스키마 버전 enum
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Getter
public enum EventSchemaVersion {

    V1_0("1.0", "버전 1.0"),
    V1_1("1.1", "버전 1.1"),;

    private final String version;
    private final String description;

}

File: src/main/java/com/example/cqrs/command/entity/event/enumerate/EventSchemaVersion.kt
package com.example.cqrs.command.entity.event.enumerate

/**
 * 이벤트 스키마 버전
 * 이벤트 데이터 구조가 변경될 때 버전 관리를 위함
 */
enum class EventSchemaVersion(
    val version: String,
    val description: String
) {
    V1_0("1.0", "기본 버전"),
    V1_1("1.1", "확장 필드 추가"),
    V2_0("2.0", "주요 구조 변경")
}
